contrib/geoToolbox.lua

```diff
@@ -0,0 +1,590 @@
+
+        if (first_elevation and second_elevation) then
+          local in_between_elevation = first_elevation + (second_elevation - first_elevation) * percent_in_between
+
+dt.print_error(first_image_date)
+dt.print_error(second_image_date)
+dt.print_error(image_date)
+dt.print_error(first_elevation)
+dt.print_error(second_elevation)
+dt.print_error(in_between_elevation)
+
+        end
+      end
+    end
+  end
+end
+
+local copyGPS_have_data = false
+local copyGPS_latitude = ''
+local copyGPS_longitude = ''
+local copyGPS_elevation = ''
+
+local function copyGPS()
+  local sel_images = dt.gui.selection()
+
+  copyGPS_latitude = ''    
+  copyGPS_longitude = ''
+  copyGPS_elevation = ''
+
+  for _,image in ipairs(sel_images) do
+    if not image then
+      copyGPS_have_data = false
+    else
+      copyGPS_have_data = true
+      if (image.latitude and labelCopyGPSlat.value) then
+        copyGPS_latitude = image.latitude    
+      end
+      if (image.longitude and labelCopyGPSlon.value) then
+          copyGPS_longitude = image.longitude
+      end
+      if (image.elevation and labelCopyGPSele.value) then
+          copyGPS_elevation = image.elevation
+      end
+    end
+
+    labelCopyGPSlat.label = "latitude: " .. copyGPS_latitude
+    labelCopyGPSlon.label = "longitude: " ..copyGPS_longitude
+    labelCopyGPSele.label = "elevation: " .. copyGPS_elevation
+
+    return
+  end
+end
+
+local function pastGPS(image)
+  local sel_images = dt.gui.selection()
+
+  for _,image in ipairs(sel_images) do
+    if (labelCopyGPSlat.value) then
+      image.latitude = copyGPS_latitude
+    end
+    if (labelCopyGPSlon.value) then    
+      image.longitude = copyGPS_longitude
+    end
+    if (labelCopyGPSele.value) then
+      image.elevation = copyGPS_elevation
+    end
+  end
+end
+
+local function openLocationInGnomeMaps()
+
+  if not df.check_if_bin_exists("gnome-maps") then
+    dt.print_error(_("gnome-maps not found"))
+    return
+  end	
+    
+  local sel_images = dt.gui.selection()
+  
+  local lat1 = 0;
+  local lon1 = 0;
+  local i = 0;
+  
+  -- Use the first image with geo information
+  for _,image in ipairs(sel_images) do
+    if ((image.longitude and image.latitude) and 
+        (image.longitude ~= 0 and image.latitude ~= 90) -- Sometimes the north-pole but most likely just wrong data
+       ) then
+        lat1 = image.latitude;
+        lon1 = image.longitude;
+        break
+    end
+
+    local startCommand
+    startCommand = "gnome-maps \"geo:" .. lat1 .. "," .. lon1 .."\""
+    dt.print_error(startCommand)
+    
+    if coroutine.yield("RUN_COMMAND", startCommand) then
+      dt.print(_("Command failed ..."))
+    end
+
+  end
+
+end
+
+
+-- I used code from here:
+-- http://stackoverflow.com/questions/27928/how-do-i-calculate-distance-between-two-latitude-longitude-points
+local function getDistance(lat1, lon1, ele1, lat2, lon2, ele2)
+
```

Comment:
When I first built the libraries I thought about adding these, but I had so many others that I saved them for another day.  :-)  I think it could be time.